# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1466091749 25200
#      Thu Jun 16 08:42:29 2016 -0700
# Node ID a53f46577e391932cbaab63a5e54a3f8c38d4c20
# Parent  43a5d019e6cc92ffa922d59ab740f75846cdc572
Fix variable expansion in dynamic linking flags for macOS (bug #48215)

* configure.ac (DL_LDFLAGS): Use ${ac_top_build_prefix} for correct value.
(MKOCTFILE_DL_LDFLAGS): Defer expansion of ${bindir} and ${version} until they
are fully defined.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2033,18 +2033,20 @@ case $canonical_host_type in
   ;;
   alpha*-dec-osf*)
     CPICFLAG=
     CXXPICFLAG=
     FPICFLAG=
     SH_LDFLAGS="-shared -Wl,-expect_unresolved -Wl,'*'"
   ;;
   *-*-darwin*)
-    DL_LDFLAGS="-bundle -bundle_loader ${top_builddir}/libinterp/octave ${LDFLAGS}"
-    MKOCTFILE_DL_LDFLAGS="-bundle -bundle_loader ${bindir}/octave-${version}${EXEEXT}"
+    DL_LDFLAGS="-bundle -bundle_loader ${ac_top_build_prefix}libinterp/octave ${LDFLAGS}"
+    ## Contains variables that are defined and undefined at this point, so use
+    ## appropriate quoting to defer expansion of ${bindir} and ${version}.
+    MKOCTFILE_DL_LDFLAGS='-bundle -bundle_loader ${bindir}/octave-${version}'"${EXEEXT}"
     SH_LDFLAGS="-dynamiclib -single_module ${LDFLAGS}"
     case $canonical_host_type in
       powerpc-*)
         CXXPICFLAG=
         CPICFLAG=
         FPICFLAG=
       ;;
     esac
@@ -3303,16 +3305,21 @@ AC_CONFIG_COMMANDS_PRE([
   eval texi_macros_file="\"$texi_macros_file\""
   eval imagedir="\"$imagedir\""
   eval octdatadir="\"$octdatadir\""
   eval startupfiledir="\"$startupfiledir\""
   eval localstartupfiledir="\"$localstartupfiledir\""
   eval man1dir="\"$man1dir\""
   eval man1ext="\"$man1ext\""
   eval infofile="\"$infofile\""
+
+  ## Also expand MKOCTFILE_DL_LDFLAGS, because it may depend on the values
+  ## of variables expanded above.
+
+  eval MKOCTFILE_DL_LDFLAGS="\"$MKOCTFILE_DL_LDFLAGS\""
 ])
 
 ### Do the substitutions in Makefiles and other scripts that
 ### hold configuration info.
 
 AUTOCONF_SUBST_VARS="`echo $ac_subst_vars`"
 
 AC_SUBST(AUTOCONF_SUBST_VARS)
